docs/reddit-migration/strategy/schema.md

```diff
@@ -0,0 +1,61 @@
+ - Django treebeard: https://django-treebeard.readthedocs.io/en/latest/
+
+Reddit stores comment trees in Cassandra, which is overkill for our current use cases. That said, we need to be able to migrate the comments in such a manner that we achieve the following:
+
+- Maintain current API result structures
+  - Specifically, we need to support some form of reddit's "more comments" pagination. This **does not** need to partition the comments exactly as reddit would, the interface just needs to be compatible with how the frontend traverses it.
+- Support currently used sorting methods (best, old, new)
+- Support rendering an arbitrary subtree of the comments for the comment detail view
+
+From the Prior Art listed above, Materialized Paths are probably our best bet and we're already familiar to some degree with them because they're what `wagtail` uses under the hood for page hierarchies.
+
+The big thing here is that in order to support the sorts we need one for *each type of sort*. That precludes using a single table for everything so we'll want to have a hierarchical table for each sort and then `prefetch_related` on the `Comment` table so that code can work generically over similarly structured tables. This approach means these sorting hierarchy tables end up being an acceleration data structure for extracting a portion of the comment tree.
+
+### Votes
+
+We should have a table for `PostVote` and `CommentVote`, these should have the following columns:
+
+- `comment = ForeignKey(Comment)` / `post = ForeignKey(Post)` - item that was voted on
+- `voter = ForeignKey(User)` - the user who made the vote
+- `direction = SmallIntegerField()` - we store either a `1` or `-1` to indicate which direction the user voted. In practice, `PostVote` will always have a `direction == 1`, but it's useful to carve this out now in case that changes at a later date. It's also useful to be able to run a `SUM()` over that column.
+
+Votes are probably the trickiest moving piece in this reddit migration because it's impractical to get the data directly from the reddit backend for a backpopulate. This probably means we're going to need either some kind of ETL process. This is only a 1-time event, so an approach as basic as an export/import of a CSV would be adequate.
+
+
+### Reported Content
+
+We need a data model for reported content (e.g. report post/comment to a moderator). We don't need much data here, the following is sufficient:
+
+- `GenericForeignKey` to `Post` / `Comment` - we should unique on this FK so that we only get one report object per content object. We should handle new reports as a `get_or_create`
+- `resolved_on = DateTimeField(null=True)` - when a report is resolved, this is set the the current datetime, nullable datetimes have useful properties that we might want to take advantage of down the road (e.g. if a post is edited, delete any reports prior to the edit time)
```

Comment:
Ah, i forgot about the ignore report action, I think that still fits into this data model though, because my expectation is that resolved reports don't show up to moderators and so subsequent reports just end up being a no-op on the already resolved report. Unless you feel like there's value in knowing _how_ a report was resolved (approved, removed, or ignored). That would just be an extra column.

Reply:
I'm not looking to add new functionality. I was just trying to understand how the data model represents the current functions.

Reply:
Actually, you're right, we need some kind of value to represent the action that was taken on the report, otherwise `ignored` vs `approved` is ambiguous since both cause no change to the post or comment, but `ignored means further reports don't show up.

Reply:
Ok, decided to go with a `reports_ignored` prop on `Post`/`Comment` and keeping the current data model as-is.